pandas/plotting/_matplotlib/converter.py

```diff
@@ -55,13 +56,43 @@ def get_pairs():
     return pairs
 
 
-def register(explicit=True):
-    # Renamed in pandas.plotting.__init__
-    global _WARN
+def pandas_converters_deco(func):
```

Comment:
Is there a reason why this needs to be a decorator? Perhaps a style preference but I would find it easier to grok if the context manager was used where required in the actual plotting methods

Reply:
No need. I mainly did it for a smaller diff / less indentation :)

Happy to go either way here.